pkg/kp/kv.go

```diff
@@ -108,7 +108,6 @@ func (s *Store) ListPods(keyPrefix string) ([]ManifestResult, time.Duration, err
 // caller's responsibility to filter out unchanged manifests.
 func (s *Store) WatchPods(keyPrefix string, quitChan <-chan struct{}, errChan chan<- error, podChan chan<- ManifestResult) {
 	defer close(podChan)
-	defer close(errChan)
```

Comment:
Makes it possible for multiple goroutines to share the same error channel without one closing it under the feet of the others.

After this, I was thinking more about how to handle channel arguments and I think that, in general, it makes more sense for the sender to create and close the channel, while passing it back to the receiver. Right now all of our APIs pass channels into the function, which leaves ownership in the hands of the caller. In that case, I think it makes more sense to avoid closing the channels you're given.